### PR TITLE
Script to convert ESO's PV forecasts from CSV to NetCDF

### LINKED ISSUE #36: Write script to convert ESO PV forecasts to NetCDF
## Detailed Description
The ESO PV forecasts are provided as a set of CSV files.

Convert these to a single NetCDF file.

* Split the ASL and ML forecasts into two separate xr.DataArrays
* Only take the 2 day forecasts

This is a sub-task of #35

## Possible Implementation
Python script

### PR DESCRIPTION
**Pull request by JackKelly**
# Pull Request

## Description

ESO provided their historical PV forecasts as almost 100 GB of CSV files.  To make use of these files, we first need to convert these CSV files to a single NetCDF file.  This script implements that conversion.

Fixes #36

## How Has This Been Tested?

This has run on `leonardo` and the resulting NetCDF file looks good.

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+* ASL: "Advanced Statistical Learning": These are hand-crafted models written in `R` by the
+       ESO forecasting team.
+* ML: "Maching Learning": ML models, trained from historical data, created by the ESO Labs team.
+
+Each DataArray has three dimensions:
+
+1. `gsp_id`: A string identifying the Grid Supply Point region.
+2. `forecast_date_time`: The UTC datetime when ESO ran their forecast.  Not exactly the
+    same as the NWP init time.  This script takes the floor('30T') of the
```
Comment (peterdudfield) [inline scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:19]: NWP?

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    df = df.rename(columns={'SITE_ID': 'gsp_id'})
+
+    # Set index
+    df = df.set_index(['gsp_id', 'forecast_date_time', 'step'])
+    series = df.squeeze()
+    del df
+
+    # Remove duplicate indicies
+    duplicated = series.index.duplicated()
+    series = series[~duplicated]
```
Comment (peterdudfield) [inline scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:163]: could you reduce it to one line using - https://pandas.pydata.org/docs/reference/api/pandas.Series.drop_duplicates.html

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            data_arrays[algo_name] = convert_to_dataarray(split_forecasts[algo_name])
+        del split_forecasts
+
+        dataset = xr.Dataset(data_arrays)
+        datasets.append(dataset)
+        del dataset, data_arrays
+
+    dataset = xr.concat(datasets, dim='forecast_date_time')
+
+    dataset.to_netcdf(DST_NETCDF)
```
Comment (peterdudfield) [inline scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:191]: can you say roughly how big this file is? If its very big, would any encoding or compression help?

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py (hunk 1 above)
Comment (JackKelly) [reply scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:19]: Sorry, it is a bit confusing!  I've added more context and detail to the docstring!

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    return eso_forecasts_df
+
+
+def split_asl_and_ml_forecasts(eso_forecasts_df: pd.DataFrame) -> dict[str, pd.DataFrame]:
+    """Split the forecasts into ASL forecasts, and ML forecasts.
+
+    Returns a dict where the keys are the ESO forecasting algorithm name 'ASL' or 'ML',
+    and the values are the pd.DataFrame of forecast data.
+    """
+    eso_forecasts_df.SCRIPT_NAME = eso_forecasts_df.SCRIPT_NAME.str.upper()
```
[the remark this answers is not in the document]
Comment (JackKelly) [reply scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:118]: Sure!  And I've added some more comments

File: scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py (hunk 3 above)
Comment (JackKelly) [reply scripts/convert-ESO-PV-forecasts-from-CSV-to-HDF5.py:191]: I've added a comment.

**Pull request merged**